sim/scenarios/helper/quic-packet.cc

```diff
@@ -6,17 +6,48 @@
 #include "ns3/packet.h"
 #include "ns3/ppp-header.h"
 #include "ns3/ipv4-header.h"
+#include "ns3/ipv6-header.h"
 #include "ns3/udp-header.h"
 
 using namespace ns3;
 using namespace std;
 
+
+bool IsUDPPacket(Ptr<Packet> p) {
+    PppHeader ppp_hdr = PppHeader();
+    p->RemoveHeader(ppp_hdr);
```

Comment:
Maybe check if PPP header and return early with error if it isn't.

Reply:
The PPP header here is just 2 bytes indicating the protocol. Not sure how I would test for it.